source/algorithms/discrete_time.cc

```diff
@@ -0,0 +1,65 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2019 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+#include <deal.II/algorithms/discrete_time.h>
+
+#include <cmath>
+#include <limits>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace Algorithms
+{
+  DiscreteTime::DiscreteTime(const double start_time,
+                             const double final_time,
+                             const double start_step_size)
+    : m_start_time{start_time}
+    , m_final_time{final_time}
+    , m_start_step_size{start_step_size}
+    , m_current_time{start_time}
+    , m_next_time{start_time + start_step_size}
+  {
+    Assert(final_time > start_time, ExcInternalError());
+    Assert(start_step_size <= final_time - start_time, ExcInternalError());
```

Comment:
Is this necessary? If there is only one time step to be had between start and end time, we'll just treat it like every other "last time step".

Reply:
Not it is not necessary. I fixed it.